### PR TITLE
Machine learning model for classifying transportation mode

### PR DESCRIPTION
**Pull request by HansBambel**
This PR adds:

- [x] Dataset class for transportation modes
- [x] Function to create train+valid and test set folders from data
- [x] Dataset are normalized using mean and std of training set
- [x] Neural Network with Convolutions (["A Convolutional Neural Network for Transportation Mode Detection Based on Smartphone Platform"](https://ieeexplore.ieee.org/abstract/document/8108764))
- [x] Using all sensors or (like in the above paper) magnitude of accelerometer with removed gravity
- [x] Working randomforest classification
- [x] Shrunk _acc_magnitude_ dataset creation from ~11-17s per file to merely 0.3s per file (thanks to numba)
- [x] Apply numba to normal dataset creation
- [ ] Smoothing of signals (postpone to later)

Added need for [numba-package](http://numba.pydata.org/). This reduces creation of magnitude calculation to only 10% of pure python implementation.

### CODE REVIEW

**Comment by HansBambel** [comment]
ROC score of Randomforest (with 100 trees) is 0.72 on the (non-corrected) blackforest dataset

**Comment by HansBambel** [comment]
Smoothing requires scipy.signal.savgol_filter to be able to converted to numba function... maybe later...
